tests/java_str.rs

```diff
@@ -0,0 +1,24 @@
+#![cfg(feature = "invocation")]
+
+mod util;
+
+use crate::util::unwrap;
+use jni::strings::JavaStr;
+use std::mem::ManuallyDrop;
+use util::attach_current_thread;
+
+const TEST_STRING: &str = "";
+
+#[test]
+fn get_java_str() {
+    let env = attach_current_thread();
+    let string = unwrap(&env, env.new_string(TEST_STRING));
+    let javastr = unwrap(&env, JavaStr::from_env(&env, string.clone()));
+
+    let result = unsafe { JavaStr::from_raw(javastr.get_raw(), string, &env) };
```

Comment:
I think perhaps the docs for `from_raw` could make it clearer that for safety `from_raw` should only be giving a raw pointer that you own and maybe show an example of using `into_raw()` and then `from_raw().

The intention with the API is that there would be symmetry between taking ownership of the raw pointer via `into_raw()` and then returning ownership with `from_raw()`.

Although it's just a test that's self-aware of the imbalance it's possible someone might look at this code for guidance about how to use the API so it could be good if we test it in a "safe" way by calling `into_raw()` before calling `from_raw()` - so then there wouldn't be a need to use `ManuallyDrop` to avoid a double free.